recommender/static/js/src/recommender.js

```diff
@@ -1,852 +1,930 @@
-					$('.recommender_vote_score', element).each(function(idx, ele){ 
-						if (parseInt($(ele).text()) < 0) {
-							pos = idx;
-							return false;
-						}
-					});
-
-					/* Show the added resource at right place (pos), based on sorting the votes, and lead student to that page */
-					if ($('.recommender_resource', element).length == 0) {
-						$('.noResourceIntro', element).addClass('hidden');
-						$('.descriptionText', element).show();
-						currentPage = 1;
-						var newDiv = $('.recommender_resourceTemplate', element).clone().removeClass('hidden').removeClass('recommender_resourceTemplate').addClass('recommender_resource');
-					}
-					else {
-						if (pos == -1) {
-							var toDiv = $('.recommender_resource:last', element);
-							currentPage = Math.ceil(($('.recommender_resource', element).length+1)/entriesPerPage);
-						}
-						else {
-							var toDiv = $('.recommender_resource:eq(' + pos.toString() + ')', element);
-							currentPage = Math.ceil((pos + 1)/entriesPerPage); 
-						}
-						var newDiv = $(toDiv).clone();
-					}
-					/* Generate the div for the new resource */
-					$(newDiv).find('.recommender_vote_arrow_up,.recommender_vote_score,.recommender_vote_arrow_down')
-						.removeClass('downvoting').removeClass('upvoting');
-					$(newDiv).find('.recommender_vote_score').text('0');
-					$(newDiv).find('a').attr('href', result['url']);
-					$(newDiv).find('a').text(result['title']);
-					$(newDiv).find('.recommender_descriptionImg').text(result['description']);
-					$(newDiv).find('.recommender_descriptionText').text(result['descriptionText']);
-					$(newDiv).find('.recommender_entryId').text(result['id']);
-					$(newDiv).find('.recommender_problematicReason').text('');
-                    $(newDiv).find('.recommender_endorse_reason').text('');
-					$(newDiv).find('.flagResource').removeClass('problematic');
-                    $(newDiv).find('.endorse').removeClass('endorsed');
-					bindEvent(newDiv);
-					if (is_user_staff) { addFunctionsForStaffPerResource(newDiv); }
-
-					if ($('.recommender_resource', element).length == 0) {
-						$('.recommender_resourceTemplate', element).before(newDiv);
-					}
-					else {
-						if (pos == -1) { $(toDiv).after(newDiv); }
-						else { $(toDiv).before(newDiv); }
-					}
-					addTooltipPerResource(newDiv);
-					addResourceReset();
-					paginationRow();
-					pagination();
-					backToView();
-				}
-				else {
-                    alert(result['error']);
+        addResourceDependentTooltip(newDiv);
+
+        return newDiv;
+    }
+    
+    /**
+     * Add the new resource to the database, and update the resource list.
+     * @param {dictionary} data The resource to be added.
+     */
+    function addResource(data) {
+        $.ajax({
+            type: "POST",
+            url: addResourceUrl,
+            data: JSON.stringify(data),
+            success: function(result) {
+                if (result['Success'] == true) {
```

Comment:
You shouldn't need to check for success in a `success` handler -- it checks the HTTP status code.